packages/datetime/src/_daterangepicker.scss

```diff
@@ -63,20 +63,45 @@ Styleguide components.datetime.daterangepicker.js
     margin-right: $pt-grid-size;
   }
 
+  .DayPicker-Day--hovered-range:not(.DayPicker-Day--outside) {
+    border-radius: 0;
+
+    // need to disable hover-styles for all variants of selected dates
+    // stylelint-disable max-line-length
+    &:not(.DayPicker-Day--selected):not(.DayPicker-Day--selected-range):not(.DayPicker-Day--selected-range-start):not(.DayPicker-Day--selected-range-end) {
```

Comment:
Selected styles are defined in another file with less-specific selectors, so that makes things tricky. Can definitely return to this in a later PR to see if there's a better approach.